lib/Mongo/MongoCollection.php

```diff
@@ -434,7 +454,18 @@ public function count($query = array())
      * @return array|boolean If w was set, returns an array containing the status of the save.
      * Otherwise, returns a boolean representing if the array was not empty (an empty array will not be inserted).
      */
-    public function save($a, array $options = array()) {}
+    public function save($a, array $options = array())
+    {
+        if (is_object($a)) {
+            $a = (array)$a;
+        }
+        if ( ! array_key_exists('_id', $a)) {
+            $a['_id'] = new \MongoId();
+        }
+        $filter = ['_id' => $a['_id']];
+        // TODO: not sure how to generate result array
+        return $this->collection->insertOne($filter, $a, ['upsert' => true]);
```

Comment:
plus it is not working somehow, not sure if the mongodb handles upsert, fighting my way with this now, I will rebase this mess when the test passes. Can't test on my local because of the vagrant box still downloading :-(